search/includes/classes/class-search.php

```diff
@@ -903,7 +903,17 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 
 		// Cache handling
 		$is_cacheable = $this->is_url_query_cacheable( $query['url'], $args );
-		$cache_key    = 'es_query_cache:' . md5( $query['url'] . wp_json_encode( $args ) ) . ':' . wp_cache_get_last_changed( self::SEARCH_CACHE_GROUP );
+		if ( $is_cacheable ) {
+			$cache_key = 'es_query_cache:' . md5( $query['url'] . $args['body'] ) . ':' . wp_cache_get_last_changed( self::SEARCH_CACHE_GROUP );
+			/**
+			 * Serve cached response right away, if available and not stale and the query is cacheable
+			 */
+			$cached_response = $is_cacheable ? wp_cache_get( $cache_key, self::SEARCH_CACHE_GROUP ) : false;
```

Comment:
`$is_cacheable` will never be false b/c of the check in L906. I think we can just go straight to the source via `wp_cache_get()`.